trace-archiver-db/src/redpanda_engine/mod.rs

```diff
@@ -0,0 +1,144 @@
+    password: Option<String>,
+    group: Option<String>,
+) -> RedpandaBuilder {
+    let broker = format!(
+        "{0}:{1}",
+        unwrap_string_or_env_var(url, "REDPANDA_URL"),
+        unwrap_num_or_env_var(port, "REDPANDA_PORT"),
+    );
+    let user = unwrap_string_or_env_var(user, "REDPANDA_USER");
+    let password = unwrap_string_or_env_var(password, "REDPANDA_PASSWORD");
+    let group = unwrap_string_or_env_var(group, "REDPANDA_CONSUMER_GROUP");
+    let mut builder = RedpandaBuilder::default();
+    builder
+        .set_bootstrap_servers(&broker)
+        .set_creation_timeout_ms(3000);
+    if !group.is_empty() {
+        builder.set_group_id(&group);
+    }
+    builder
+}
+
+pub async fn create_topic(builder: &RedpandaBuilder, topic: &str) -> Result<()> {
+    let admin = builder.build_admin_client().await?;
+    Ok(admin.create_topic(topic, 1, 1).await?)
+}
+
+pub fn new_consumer(builder: &RedpandaBuilder, topic: &str) -> Option<RedpandaConsumer> {
+    let consumer = builder
+        .build_consumer()
+        .unwrap_or_else(|e| log_then_panic_t(format!("Cannot create consumer : {e}")));
+    if let Err(e) = consumer.subscribe(&[topic]) {
+        if let KafkaError::Subscription(str) = e {
+            if str.eq_ignore_ascii_case(&format!("Invalid topic name {topic}")) {
+                log::info!("Topic: {topic} not found.");
+                return None;
+            } else {
+                log_then_panic(format!("Cannot subscribe to topic : {str}"));
+            }
+        } else {
+            log_then_panic(format!("Subscription error : {e}"));
+        }
+    }
+    Some(consumer)
+}
+
+pub async fn consumer_recv(consumer: &RedpandaConsumer) -> Result<BorrowedMessage> {
+    match consumer.recv().await {
+        Err(e) => Err(e.into()), //log::warn!("Kafka error: {}", e),
+        Ok(message) => match message.payload() {
+            Some(payload) => match digitizer_analog_trace_message_buffer_has_identifier(payload) {
+                true => Ok(message),
+                false => Err(anyhow!(
+                    "Unexpected message type on topic \"{}\"",
+                    message.topic()
+                )),
+            },
+            None => Err(anyhow!(
+                "No payload found in message type on topic \"{}\"",
+                message.topic()
+            )),
+        },
+    }
+}
+
+pub fn extract_payload<'a, 'b: 'a>(
+    message: &'b BorrowedMessage<'b>,
+) -> Result<DigitizerAnalogTraceMessage<'a>> {
+    let payload = message.payload().unwrap();
+    if !digitizer_analog_trace_message_buffer_has_identifier(payload) {
+        return Err(anyhow!("Unexpected message"));
+    }
+    match root_as_digitizer_analog_trace_message(payload) {
+        Ok(data) => {
+            log::info!(
+                "Trace packet: dig. ID: {}, metadata: {:?}",
+                data.digitizer_id(),
+                data.metadata()
+            );
+            /*metrics::MESSAGES_RECEIVED
```

Comment:
This code can be removed.

Reply:
Do you mean the commented out code or the whole log::info! block?

Reply:
The commented out code, or more specifically things relating to the `metrics` module.

Background is that originally all the tools were instrumented with OpenMetrics flavour metrics with the intention that at some point they would be monitored. In reality that has not yet happened and it is not clear if that will happen or if this will happen via EPICS instead.